### PR TITLE
Parametric maps

### PR DESCRIPTION
**Pull request by lheagy**
- add a parametric block map (2D and 3D) (eg. as shown in https://github.com/simpeg-research/heagy_2018_AEM/blob/master/Heagy_etal_2018_aem_workshop.pdf) 

image

- rename Parametrized maps to Parametric maps

### CODE REVIEW

**Comment by lheagy** [comment]
@sgkang: would you be willing to take a look at this?

**Comment by lheagy** [comment]
@sgkang: would you be good if I merge this?

File: SimPEG/Maps.py [hunk 1]
```diff
@@ -1871,9 +1893,306 @@ def deriv(self, m):
             ]).T)
 
 
-class ParametrizedCasingAndLayer(ParametrizedLayer):
+class ParametricBlock(BaseParametric):
+    """
+        Parametric Block in a Layered Space
```
Comment (sgkang) [inline SimPEG/Maps.py]: is this in layered earth? 
I thought you are assuming homogenous earth (or background is layered-earth here?)

Comment (lheagy) [reply SimPEG/Maps.py]: ah, thanks that is an old doc-string that I copied from the Parametric layer map

**Pull request merged**